### PR TITLE
VectorTools::interpolate changed to allow FE_Nothing

### PR DESCRIPTION
**Pull request by angelrca**
VectorTools::interpolate does not support the use of hp::fe_collections containing FE_Nothing scalar finite elements.
There is an assert which checks the number of support points of the finite elements and it fails when fe_nothing is read. This condition can be modified to allow the use of finite elements with zero degrees of freedom (FE_Nothing).
In the loop over the cells a condition is added so the interpolation is only made on the finite elements with degree not equal to zero.

### CODE REVIEW

**Comment by Rombur** [comment]
Can you add the test that you sent on the mailing list and a line in the specific improvement section of  doc/news/changes.h that says that VectorTools::interpolate now works with FE_Nothing. Lastly, you need to run astyle to indent the code correctly:
```
astyle --options=<SOURCE DIRECTORY>/contrib/utilities/astyle.rc <file>
``` 
If you are on cluster you can download astyle here: http://sourceforge.net/projects/astyle/files/ otherwise you should be able to get it through your package manager.

**Comment by angelrca** [comment]
I have used astyle to indent the file, but now the indentation fails in parts of the code which i have not changed. should i use the astyle command with another additional option?
i'm a new using these tools so maybe i'm doing something wrong

**Comment by Rombur** [comment]
Maybe you should try to use astyle 2.4. This is the version that we use.

**Comment by angelrca** [comment]
Yes, that did the trick.
The test case and changes.h are uploaded

File: doc/news/changes.h [hunk 1]
```diff
@@ -980,6 +980,11 @@ inconvenience this causes.
   <br>
   (David Wells, 2015/04/13)
   </li>
+
+  <li> Improved: VectorTools::interpolate now works with FE_Nothing.
+  <br>
+  (Angel Rodriguez, 2015/06/03)
+  </li>
```
Comment (Rombur) [inline doc/news/changes.h]: This should actually go at the top of the section. Could you also move the comment above yours.

File: tests/hp/interpolate_nothing_03.cc [hunk 2]
```diff
@@ -0,0 +1,99 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2014 by the deal.II authors
```
Comment (Rombur) [inline tests/hp/interpolate_nothing_03.cc]: Change the date to 2015.

**Comment by Rombur** [comment]
Except the two small changes, it looks good. Thanks for the patch !!

**Comment by bangerth** [comment]
This broke one of our testcases, as shown in #1015 (FE_DGQ has no support points and degree==0, but is a valid element anyway).

**Pull request merged**